projects/vdk-core/plugins/vdk-plugin-control-cli/src/vdk/plugin/control_cli_plugin/execution_skip.py

```diff
@@ -0,0 +1,164 @@
+from vdk.internal.builtin_plugins.config.job_config import JobConfigKeys
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.internal.builtin_plugins.termination_message.action import WriteToFileAction
+from vdk.internal.builtin_plugins.termination_message.writer_configuration import (
+    WriterConfiguration,
+)
+from vdk.internal.control.auth.auth import Authentication
+from vdk.internal.control.configuration.vdk_config import VDKConfig
+from vdk.internal.control.rest_lib.factory import ApiClientFactory
+
+log = logging.getLogger(__name__)
+
+
+class ConcurrentExecutionChecker:
+    def __init__(
+        self, rest_api_url, api_token, api_token_authorization_url, auth_type
+    ) -> None:
+        ConcurrentExecutionChecker._authenticate(
+            api_token_authorization_url, api_token, auth_type
+        )
+        self.execution_api_client = ApiClientFactory(rest_api_url).get_execution_api()
+
+    def is_job_execution_running(
+        self, job_name, job_team, job_execution_attempt_id
+    ) -> bool:
+        """
+        :param job_name : str value for the job name
+        :param job_team : str value for the job team
+        :param job_execution_attempt_id : str current job's execution id as returned by VDK execution_state
+        This method calls the "/data-jobs/for-team/{team_name}/name/{job_name}/executions"
+        endpoint with the additional params "submitted" and "running". The API will return all jobs that are
+        submitted and running with the provided team and job name. We have the following possible scenarios:
+        API returns an empty list - this means no other data jobs excecutions are running and we can proceed with current execution.
+        API returns one execution with a different ID - this means that another excecution is running and we should skip.
+        API returns one job with the same ID - this means that we can proceed with excecution.
+        API returns two jobs one of which has a different ID - we need to skip current excecution.
+        It then parses the response and returns a boolean.
+        :return: bool
+                 If the data job has a different execution running in K8S.
+                 A running different execution means that there is a data job execution currently
+                 submitted or running with a different id and in that case we should skip current execution
+                 until the other finishes.
+        """
+        jobs = self._get_running_job_executions(job_name, job_team)
+        job_execution_running = False
+        for job_execution in jobs:
+            # We break only if we have a job with a different execution id
+            # Sometimes ids returned by VDK execution_state can slightly differ than
+            # ids returned by the API. VDK ids  have '-xxxxx' appended at the end,
+            # where x is a random alphameric char. Otherwise ids start the same.
```

Comment:
nit: alphanumeric*

Reply:
It's am alternative form for the same word, though alphanumeric is used more often.